webapp/src/components/modals/create_issue/create_issue.jsx

```diff
@@ -0,0 +1,221 @@
+    error: null,
+    repoValue: '',
+    issueTitle: '',
+    issueDescription: '',
+    showErrors: false,
+    issueTitleValid: true,
+};
+
+export default class CreateIssueModal extends PureComponent {
+    static propTypes = {
+        close: PropTypes.func.isRequired,
+        create: PropTypes.func.isRequired,
+        post: PropTypes.object,
+        theme: PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = initialState;
+        this.validator = new Validator();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (this.props.post && !prevProps.post) {
+            this.setState({issueDescription: this.props.post.message}); //eslint-disable-line react/no-did-update-set-state
+        }
+    }
+
+    // handle issue creation after form is populated
+    handleCreate = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+
+        if (!this.validator.validate() || !this.state.issueTitle) {
+            this.setState({
+                issueTitleValid: Boolean(this.state.issueTitle),
+                showErrors: true,
+            });
+            return;
+        }
+
+        const issue = {
+            title: this.state.issueTitle,
+            body: this.state.issueDescription,
+            repo: this.state.repoValue,
+            post_id: this.props.post.id,
+        };
+
+        this.setState({submitting: true});
+
+        this.props.create(issue).then((created) => {
+            if (created.error) {
+                this.setState({
+                    error: created.error.message,
+                    showErrors: true,
+                    submitting: false});
+                return;
+            }
+            this.handleClose(e);
+        });
+    };
+
+    handleClose = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+        this.setState(initialState, this.props.close);
+    };
+
+    handleRepoValueChange = (name) => {
+        this.setState({
+            repoValue: name,
+        });
+    };
+
+    handleIssueTitleChange = (newValue) => {
+        this.setState({
+            issueTitle: newValue,
+        });
+    };
+
+    handleIssueDescriptionChange = (newValue) => {
+        this.setState({
+            issueDescription: newValue,
+        });
+    };
+
+    render() {
+        const {visible, theme} = this.props;
+        const {error, submitting} = this.state;
+        const style = getStyle(theme);
+
+        const requiredMsg = 'This field is required.';
+        let issueTitleValidationError = null;
+        if (this.state.showErrors && !this.state.issueTitleValid) {
+            issueTitleValidationError = (
+                <p className='help-text error-text'>
+                    <span>{requiredMsg}</span>
+                </p>
+            );
+        }
+
+        if (!visible) {
+            return null;
```

Comment:
I think it makes more sense to have this at the top of the function

Reply:
Fixed in eab01f5